processing/src/main/java/io/druid/query/aggregation/post/ExpressionPostAggregator.java

```diff
@@ -36,25 +37,22 @@
 import java.util.Objects;
 import java.util.Set;
 
-/**
- */
 public class ExpressionPostAggregator implements PostAggregator
 {
-  private static final Comparator<Number> DEFAULT_COMPARATOR = new Comparator<Number>()
-  {
-    @Override
-    public int compare(Number o1, Number o2)
-    {
-      if (o1 instanceof Long && o2 instanceof Long) {
-        return Long.compare(o1.longValue(), o2.longValue());
-      }
-      return Double.compare(o1.doubleValue(), o2.doubleValue());
-    }
-  };
+  private static final Comparator<Comparable> DEFAULT_COMPARATOR = Comparator.nullsFirst(
+      (Comparable o1, Comparable o2) -> {
+        if (o1 instanceof Long && o2 instanceof Long) {
```

Comment:
I guess Druid isn't really consistent here. Sometimes nulls are treated as zeros (like when reading from nonexistent columns) and sometimes they are treated as less than any other number (like in most comparators).

My sense is that most comparators in Druid treat nulls as less than any other number (and actually return them to the user as `null` not as `0`) so I am inclined to keep this the way it is.